### PR TITLE
checking csrf_protection in delete method.

### PR DESCRIPTION
**Pull request by oleg-andreyev**
## Subject

There are cases when I can disable `csrf_protection` via `$formOptions`, but when I call deleted method this option is ignored, because `deleteAction` does not use `Form`

Ideally when `deleteAction` should use "delete" form (empty form)

I am targeting this branch, because feature.

## Changelog

```markdown
### Changed
- deleteAction now respects csrf_protection
```

## To do
    
- [x] Update the tests

### CODE REVIEW

File: src/Controller/CRUDController.php [hunk 1]
```diff
@@ -1418,6 +1418,11 @@ protected function getAclRoles()
      */
     protected function validateCsrfToken($intention)
     {
+        $csrfProtection = $this->admin->getFormBuilder()->getOption('csrf_protection');
+        if ($csrfProtection === false) {
```
Comment (greg0ire) [inline src/Controller/CRUDController.php]:
```suggestion
        if ($this->admin->getFormBuilder()->getOption('csrf_protection') === false) {
```

File: src/Controller/CRUDController.php [hunk 2]
```diff
@@ -1418,6 +1418,11 @@ protected function getAclRoles()
      */
     protected function validateCsrfToken($intention)
     {
+        $csrfProtection = $this->admin->getFormBuilder()->getOption('csrf_protection');
```
Comment (greg0ire) [inline src/Controller/CRUDController.php]:
```suggestion
```

File: tests/Controller/CRUDControllerTest.php [hunk 3]
```diff
@@ -1465,6 +1566,45 @@ public function testDeleteActionInvalidCsrfToken(): void
         }
     }
 
+    public function testDeleteActionWithDisabledCsrfProtection()
```
Comment (greg0ire) [inline tests/Controller/CRUDControllerTest.php]:
```suggestion
    public function testDeleteActionWithDisabledCsrfProtection(): void
```

File: tests/Controller/CRUDControllerTest.php [hunk 4]
```diff
@@ -1171,6 +1173,17 @@ public function testDeleteActionAjaxSuccess1(): void
...
+
+        $this->admin->expects($this->once())
+            ->method('getFormBuilder')
+            ->willReturn($formBuilder);
+
+        $formBuilder->expects($this->once())
+            ->method('getOption')
+            ->with('csrf_protection')
+            ->willReturn(true);
+
```
Comment (greg0ire) [inline tests/Controller/CRUDControllerTest.php]: Have you considered doing this in `setUp()`? Things are getting a bit repetitive…

File: tests/Controller/CRUDControllerTest.php (hunk 3 above)
Comment (oleg-andreyev) [reply tests/Controller/CRUDControllerTest.php]: done

File: tests/Controller/CRUDControllerTest.php (hunk 4 above)
Comment (oleg-andreyev) [reply tests/Controller/CRUDControllerTest.php]: Moved to `setUp` what was possible.

**Comment by greg0ire** [comment]
There is a tiny CS issue to fix

File: tests/Controller/CRUDControllerTest.php [hunk 5]
```diff
@@ -1465,6 +1523,39 @@ public function testDeleteActionInvalidCsrfToken(): void
...
+        $this->formBuilder->expects($this->once())
+            ->method('getOption')
+            ->with('csrf_protection')
+            ->willReturn(false);
+
+        $this->admin->expects($this->once())
+            ->method('toString')
+            ->with($object)
+            ->willReturn('\stdClass');
```
Comment (phansys) [inline tests/Controller/CRUDControllerTest.php]:
```suggestion
            ->willReturn(\stdClass::class);
```

File: tests/Controller/CRUDControllerTest.php [hunk 6]
```diff
@@ -164,6 +171,12 @@ protected function setUp(): void
         $this->parameters = [];
         $this->template = '';
 
+
```
Comment (phansys) [inline tests/Controller/CRUDControllerTest.php]:
```suggestion
```

Comment (oleg-andreyev) [reply tests/Controller/CRUDControllerTest.php]: done

File: tests/Controller/CRUDControllerTest.php (hunk 5 above)
Comment (oleg-andreyev) [reply tests/Controller/CRUDControllerTest.php]: it should.

**Comment by core23** [comment]
Can you please rebase @oleg-andreyev ?

**Comment by oleg-andreyev** [comment]
@core23 done.

**Comment by VincentLanglet** [comment]
Hi ! This PR seems to introduce a bug, cf https://github.com/sonata-project/SonataAdminBundle/issues/5915

Isn't another way to get the `csrf_protection` option without calling the configureFormField ?
Having `$this->getSubject()` possibly null in the configureFormField seems wrong to me.

**Pull request merged**